### PR TITLE
gui/lang: Improve dialog for some lang strings.

### PR DESCRIPTION
**Pull request by Croden1999**
- gui/lang: Update lang strings in Settings Dialog and Initial Setup. (this PR is based #2085)
- script/update-windows: Bump to 0.1.9.
- Other fixes and translation update.

Thanks @HuanJiCanShang for translation, please devs review this PR.

### CODE REVIEW

**Comment by Croden1999** [comment]
ready for review

File: vita3k/gui/src/gui.cpp [hunk 1]
```diff
@@ -63,19 +63,11 @@ void draw_info_message(GuiState &gui, EmuEnvState &emuenv) {
         ImGui::SetNextWindowPos(ImVec2(emuenv.viewport_pos.x + (display_size.x / 2) - (WINDOW_SIZE.x / 2.f), emuenv.viewport_pos.y + (display_size.y / 2.f) - (WINDOW_SIZE.y / 2.f)), ImGuiCond_Always);
         ImGui::PushStyleVar(ImGuiStyleVar_ChildRounding, 10.f * SCALE.x);
         ImGui::BeginChild("##info", WINDOW_SIZE, true, ImGuiWindowFlags_NoSavedSettings | ImGuiWindowFlags_NoDecoration);
-        auto title = fmt::format("{}", spdlog::level::to_string_view(gui.info_message.level));
-        title[0] = std::toupper(title[0]);
-        ImGui::SetWindowFontScale(RES_SCALE.x);
-        ImGui::SetCursorPosX((ImGui::GetWindowWidth() - ImGui::CalcTextSize(title.c_str()).x) / 2);
-        ImGui::TextColored(GUI_COLOR_TEXT_TITLE, "%s", title.c_str());
-        ImGui::Spacing();
-        ImGui::Separator();
```
Comment (Macdu) [inline vita3k/gui/src/gui.cpp:72]: Is there a reason for deleting this? Because it can't be translated?

Comment (Croden1999) [reply vita3k/gui/src/gui.cpp:72]: It looks more aesthetically pleasing, i did not know right or not

image

File: vita3k/gui/src/settings_dialog.cpp [hunk 2]
```diff
@@ -447,22 +448,20 @@ void draw_settings_dialog(GuiState &gui, EmuEnvState &emuenv) {
...
     ImGui::SetWindowFontScale(0.7f * RES_SCALE.x);
-    auto settings_str = lang["title"].c_str();
+    auto settings_str = lang.main_window["title"].c_str();
     const auto title = is_custom_config ? fmt::format("{}: {} [{}]", settings_str, get_app_index(gui, emuenv.app_path)->title, emuenv.app_path) : settings_str;
     ImGui::SetCursorPosX((ImGui::GetWindowSize().x / 2.f) - (ImGui::CalcTextSize(title.c_str()).x / 2.f));
     ImGui::TextColored(GUI_COLOR_TEXT_TITLE, "%s", title.c_str());
-    ImGui::PopFont();
     ImGui::Spacing();
     ImGui::Separator();
-    ImGui::SetWindowFontScale(1.f * RES_SCALE.x);
```
Comment (Macdu) [inline vita3k/gui/src/settings_dialog.cpp:457]: Why do you remove the current font? It honestly look a lot better with the vita font (at least from my point of view).
Also, this is not related to the topic of this PR.

File: vita3k/gui/src/settings_dialog.cpp [hunk 3]
```diff
@@ -447,22 +448,20 @@ void draw_settings_dialog(GuiState &gui, EmuEnvState &emuenv) {
     ImGui::Begin("##settings", &settings_dialog, ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoSavedSettings);
     ImGui::PushFont(gui.vita_font);
     ImGui::SetWindowFontScale(0.7f * RES_SCALE.x);
```
Comment (Croden1999) [inline vita3k/gui/src/settings_dialog.cpp:450]: Mr. @Macdu , is exist right here

File: vita3k/gui/src/settings_dialog.cpp [hunk 4]
```diff
@@ -1191,20 +1189,18 @@ void draw_settings_dialog(GuiState &gui, EmuEnvState &emuenv) {
...
-    if (ImGui::Button(is_apply ? (is_reboot ? lang["save_reboot"].c_str() : lang["save_apply"].c_str()) : lang["save"].c_str(), BUTTON_SIZE)) {
+    if (ImGui::Button(is_apply ? (is_reboot ? lang.main_window["save_reboot"].c_str() : lang.main_window["save_apply"].c_str()) : lang.main_window["save"].c_str(), BUTTON_SIZE)) {
         save_config(gui, emuenv);
         if (is_apply)
             set_config(gui, emuenv, emuenv.io.app_path);
     }
     if (ImGui::IsItemHovered())
-        ImGui::SetTooltip("%s", lang["keep_changes"].c_str());
+        ImGui::SetTooltip("%s", lang.main_window["keep_changes"].c_str());
     ImGui::PopFont();
```
Comment (Croden1999) [inline vita3k/gui/src/settings_dialog.cpp:1204]: and here

File: vita3k/gui/src/settings_dialog.cpp (hunk 2 above)
Comment (Croden1999) [reply vita3k/gui/src/settings_dialog.cpp:457]: if not touch this, here looks like text is too big

**Comment by Macdu** [comment]
@Croden1999 
Without your PR:
image
With your PR:
image
The role of this PR shouldn't be to change the font used by default in the settints.

**Comment by Croden1999** [comment]
> @Croden1999 Without your PR: image With your PR: image The role of this PR shouldn't be to change the font used by default in the settings.

If not change this line, here is display the mono font.
Only title and main button used vita font and also default font.

File: vita3k/gui/src/gui.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/gui/src/gui.cpp]: Please, don't delete this, I'm put here for a good reason, it's to indicate whether it's an error or information

File: vita3k/gui/src/about_dialog.cpp [hunk 5]
```diff
@@ -71,6 +71,10 @@ void draw_about_dialog(GuiState &gui, EmuEnvState &emuenv) {
 
     ImGui::Text("%s", lang["vita3k"].c_str());
     ImGui::Spacing();
+#if defined(__ANDROID__)
+    ImGui::TextWrapped("%s", lang["for_android"].c_str());
+    ImGui::Spacing();
+#endif
```
Comment (Zangetsu38) [inline vita3k/gui/src/about_dialog.cpp:77]: Last time i have put code ,for android my pr have get killed with said: Don't add code for android in the master.

File: vita3k/gui/src/gui.cpp [hunk 6]
```diff
@@ -63,8 +63,7 @@ void draw_info_message(GuiState &gui, EmuEnvState &emuenv) {
         ImGui::SetNextWindowPos(ImVec2(emuenv.viewport_pos.x + (display_size.x / 2) - (WINDOW_SIZE.x / 2.f), emuenv.viewport_pos.y + (display_size.y / 2.f) - (WINDOW_SIZE.y / 2.f)), ImGuiCond_Always);
         ImGui::PushStyleVar(ImGuiStyleVar_ChildRounding, 10.f * SCALE.x);
         ImGui::BeginChild("##info", WINDOW_SIZE, true, ImGuiWindowFlags_NoSavedSettings | ImGuiWindowFlags_NoDecoration);
-        auto title = fmt::format("{}", spdlog::level::to_string_view(gui.info_message.level));
-        title[0] = std::toupper(title[0]);
+        auto title = gui.lang.app_context.main["information"];
```
Comment (Zangetsu38) [inline vita3k/gui/src/gui.cpp:66]: Revert this, is not string informations att all
this header of dialog indicate gui.info_message.level, relate level error get

File: vita3k/gui/src/welcome_dialog.cpp [hunk 7]
```diff
@@ -28,55 +28,62 @@ void draw_welcome_dialog(GuiState &gui, EmuEnvState &emuenv) {
...
     ImGui::SetWindowFontScale(RES_SCALE.x);
     ImGui::PopStyleColor();
     ImGui::Spacing();
-    ImGui::TextColored(GUI_COLOR_TEXT, "%s", lang["line_first"].c_str());
+#if defined(__ANDROID__)
+    ImGui::TextWrapped("%s", lang["for_android"].c_str());
+    ImGui::Spacing();
+    ImGui::Separator();
+    ImGui::Spacing();
+#endif
```
Comment (Zangetsu38) [inline vita3k/gui/src/welcome_dialog.cpp:44]: Same before 🤔

File: vita3k/script/update-windows.bat [hunk 8]
```diff
@@ -9,7 +9,7 @@ echo ============================================================
 
 for /f "delims=" %%g in ('powershell "((Invoke-RestMethod https://api.github.com/repos/Vita3K/Vita3K/releases/latest -timeout 2).body.Split("\"`n"\") | Select-String -Pattern 'Vita3K Build:') -replace  'Vita3K Build: '"') do @set git_version=%%g
 if exist Vita3K.exe (
-   for /f "delims=" %%v in ('powershell "((Get-Item Vita3K.exe).VersionInfo.FileVersion) -replace '0.1.8.'"') do @set version=%%v
+   for /f "delims=" %%v in ('powershell "((Get-Item Vita3K.exe).VersionInfo.FileVersion) -replace '0.1.9.'"') do @set version=%%v
```
Comment (Zangetsu38) [inline vita3k/script/update-windows.bat:12]: oupsy, totalty forget update it

File: vita3k/gui/src/gui.cpp (hunk 6 above)
Comment (Croden1999) [reply vita3k/gui/src/gui.cpp:66]: but the spdlog is display in log, should be not display in gui, if u do not like it, i revert it, Mr. Zangetsu

File: vita3k/gui/src/about_dialog.cpp (hunk 5 above)
Comment (Croden1999) [reply vita3k/gui/src/about_dialog.cpp:77]: we have good reason now,  there is only text here

**Comment by Macdu** [comment]
> Only title and main button used vita font/default font in master version, sir. If not add `PushFont(gui.vita_font)` in dialog, here is still display the monospace font, the utf-8 text cannot be displayed (utf-8 characters will be display "?") . image

Ok, I get what you are saying. But the vita font honestly looks a lot better compared to the default one in the settings. Would it be possible to pop the font only when using an asiatic language ?

File: vita3k/gui/src/gui.cpp (hunk 6 above)
Comment (Zangetsu38) [reply vita3k/gui/src/gui.cpp]: Yes, I know, it's a short word like "info", it's just to quickly tell the user in the header if "info" = good, "error" = bad, don't forget not, this dialog, totally replace the log

Sorry for this

File: vita3k/gui/src/about_dialog.cpp (hunk 5 above)
Comment (Macdu) [reply vita3k/gui/src/about_dialog.cpp:77]: I do agree that maybe this PR is not the best to add the android lines, even more when considering that some of them will be removed when merging the android version. Also it is problematic to add some of these lines to the lang struct because it would make it a lot easier to remove them.
I think you can remove the android part (sorry about that) in this PR and we can talk about it later.

File: vita3k/gui/src/app_context_menu.cpp [hunk 9]
```diff
@@ -468,11 +468,11 @@ void draw_app_context_menu(GuiState &gui, EmuEnvState &emuenv, const std::string
...
                 if (ImGui::MenuItem("Live Area", nullptr, &gui.vita_area.live_area_screen))
                     open_live_area(gui, emuenv, app_path);
-                if (ImGui::MenuItem("Search", nullptr))
+                if (ImGui::MenuItem(lang.main["search"].c_str(), nullptr))
                     open_search(APP_INDEX->title);
-                if (fs::exists(MANUAL_PATH) && !fs::is_empty(MANUAL_PATH) && ImGui::MenuItem("Manual", nullptr))
+                if (fs::exists(MANUAL_PATH) && !fs::is_empty(MANUAL_PATH) && ImGui::MenuItem(lang.main["manual"].c_str(), nullptr))
                     open_manual(gui, emuenv, app_path);
-                if (ImGui::MenuItem("Update"))
+                if (ImGui::MenuItem(lang.main["update"].c_str()))
```
Comment (Zangetsu38) [inline vita3k/gui/src/app_context_menu.cpp:475]: Sorry, for this is like i have explain you before, keep english word
(Size of button is fix) so no can get word more lenght
Is for that i have no translate it

File: vita3k/gui/src/gui.cpp [hunk 10]
```diff
@@ -63,7 +63,7 @@ void draw_info_message(GuiState &gui, EmuEnvState &emuenv) {
         ImGui::SetNextWindowPos(ImVec2(emuenv.viewport_pos.x + (display_size.x / 2) - (WINDOW_SIZE.x / 2.f), emuenv.viewport_pos.y + (display_size.y / 2.f) - (WINDOW_SIZE.y / 2.f)), ImGuiCond_Always);
         ImGui::PushStyleVar(ImGuiStyleVar_ChildRounding, 10.f * SCALE.x);
         ImGui::BeginChild("##info", WINDOW_SIZE, true, ImGuiWindowFlags_NoSavedSettings | ImGuiWindowFlags_NoDecoration);
-        auto title = fmt::format("{}", spdlog::level::to_string_view(gui.info_message.level));
+        auto title = fmt::format(fmt::runtime(gui.info_message.title), spdlog::level::to_string_view(gui.info_message.level));
         title[0] = std::toupper(title[0]);
```
Comment (Zangetsu38) [inline vita3k/gui/src/gui.cpp:67]:
```suggestion
        auto title = fmt::format("{}", gui.info_message.title);
```
for this, after more look your code, you can remove level, sorry again
You have good works for can set all langue for error and info 
so you can just keep only like i suggest without std::toupper after
level value just keep used for log only so

File: vita3k/gui/src/main_menubar.cpp [hunk 11]
```diff
@@ -93,27 +95,28 @@ static void draw_emulation_menu(GuiState &gui, EmuEnvState &emuenv) {
...
+    if (ImGui::BeginMenu(lang["title"].c_str())) {
+        ImGui::MenuItem(lang["threads"].c_str(), nullptr, &state.threads_dialog);
+        ImGui::MenuItem(lang["semaphores"].c_str(), nullptr, &state.semaphores_dialog);
+        ImGui::MenuItem(lang["mutexes"].c_str(), nullptr, &state.mutexes_dialog);
+        ImGui::MenuItem(lang["lightweight_mutexes"].c_str(), nullptr, &state.lwmutexes_dialog);
+        ImGui::MenuItem(lang["condition_variables"].c_str(), nullptr, &state.condvars_dialog);
+        ImGui::MenuItem(lang["lightweight_condition_variables"].c_str(), nullptr, &state.lwcondvars_dialog);
+        ImGui::MenuItem(lang["event_flags"].c_str(), nullptr, &state.eventflags_dialog);
+        ImGui::MenuItem(lang["memory_allocations"].c_str(), nullptr, &state.allocations_dialog);
+        ImGui::MenuItem(lang["disassembly"].c_str(), nullptr, &state.disassembly_dialog);
```
Comment (Zangetsu38) [inline vita3k/gui/src/main_menubar.cpp:109]: for this, i no know 🤔 , what do you thinks of this @Macdu ?
For me, if i have never wanted translated it, is because is literally only for dev, so no idea if is needed, but so i ask your opinion

Comment (Macdu) [reply vita3k/gui/src/main_menubar.cpp:109]: I do agree that translating this feels very useless from my point of view (I think at most 4 people use this menu, myself included).
But if someone else went through the trouble of adding language support for these menus, I don't really see a reason not to accept it.

File: vita3k/gui/src/main_menubar.cpp [hunk 12]
```diff
@@ -26,10 +26,12 @@
 namespace gui {
 
 static void draw_file_menu(GuiState &gui, EmuEnvState &emuenv) {
-    auto lang = gui.lang.main_menubar.file;
+    auto &lang = gui.lang.main_menubar.file;
     if (ImGui::BeginMenu(lang["title"].c_str())) {
         if (ImGui::MenuItem(lang["open_pref_path"].c_str()))
             open_path(emuenv.pref_path.string());
+        if (ImGui::MenuItem(lang["open_texture_folder"].c_str()))
```
Comment (nishinji) [inline vita3k/gui/src/main_menubar.cpp:33]: Please add a check to see if the folder exists

Comment (Croden1999) [reply vita3k/gui/src/main_menubar.cpp:33]: i think to add texture dir in app init.cpp? but i do not want add in this pr

File: vita3k/gui/src/app_context_menu.cpp (hunk 9 above)
Comment (Croden1999) [reply vita3k/gui/src/app_context_menu.cpp:475]: but it is in app_context.cpp, no add in live_area.cpp, i knew here is icon

File: vita3k/gui/src/main_menubar.cpp (hunk 12 above)
Comment (nishinji) [reply vita3k/gui/src/main_menubar.cpp:33]: I'm not sure what you mean, but if the texture folder doesn't exist, an error will occur.

Comment (Croden1999) [reply vita3k/gui/src/main_menubar.cpp:33]: Okay, let's put it another way. In master, only enable export texture to running games create this dir. Mr. @Macdu could you check this, wanna add texture dir in app init

**Comment by Croden1999** [comment]
> Ok, I get what you are saying. But the vita font honestly looks a lot better compared to the default one in the settings. Would it be possible to pop the font only when using an asiatic language ?

actually, i added in previous PR, just using the vita font
https://github.com/Vita3K/Vita3K/pull/1956

File: vita3k/gui/src/app_context_menu.cpp (hunk 9 above)
Comment (Zangetsu38) [reply vita3k/gui/src/app_context_menu.cpp:475]: Oh 🗿 sorry forget lol

File: vita3k/gui/src/about_dialog.cpp (hunk 5 above)
Comment (Croden1999) [reply vita3k/gui/src/about_dialog.cpp:77]: okay,i got it

File: vita3k/gui/src/main_menubar.cpp [hunk 13]
```diff
@@ -26,10 +26,16 @@
...
 static void draw_file_menu(GuiState &gui, EmuEnvState &emuenv) {
-    auto lang = gui.lang.main_menubar.file;
+    const auto TEXTURE_PATH{ emuenv.shared_path / "textures" };
+
+    auto &lang = gui.lang.main_menubar.file;
     if (ImGui::BeginMenu(lang["title"].c_str())) {
         if (ImGui::MenuItem(lang["open_pref_path"].c_str()))
             open_path(emuenv.pref_path.string());
+        if (ImGui::MenuItem(lang["open_texture_folder"].c_str())) {
+            if (fs::exists(TEXTURE_PATH))
```
Comment (nishinji) [inline vita3k/gui/src/main_menubar.cpp:36]: Why is the order reversed?

Comment (Croden1999) [reply vita3k/gui/src/main_menubar.cpp:36]: You mean fs::exist needs to before the menu item?

but this line just to test, need create dir in app init of a new PR

Comment (nishinji) [reply vita3k/gui/src/main_menubar.cpp:36]: ah ok
i think reverse would be more correct anyway

Comment (Zangetsu38) [reply vita3k/gui/src/main_menubar.cpp:36]: > You mean fs::exist needs to before the menu item? but this line just to test, need create dir in app init of a new PR this line will be delete

Yes, all time write check before menu 🫡, 1nd set both is same line with &&

File: vita3k/gui/src/gui.cpp [hunk 14]
```diff
@@ -63,8 +63,7 @@ void draw_info_message(GuiState &gui, EmuEnvState &emuenv) {
         ImGui::SetNextWindowPos(ImVec2(emuenv.viewport_pos.x + (display_size.x / 2) - (WINDOW_SIZE.x / 2.f), emuenv.viewport_pos.y + (display_size.y / 2.f) - (WINDOW_SIZE.y / 2.f)), ImGuiCond_Always);
         ImGui::PushStyleVar(ImGuiStyleVar_ChildRounding, 10.f * SCALE.x);
         ImGui::BeginChild("##info", WINDOW_SIZE, true, ImGuiWindowFlags_NoSavedSettings | ImGuiWindowFlags_NoDecoration);
-        auto title = fmt::format("{}", spdlog::level::to_string_view(gui.info_message.level));
-        title[0] = std::toupper(title[0]);
+        auto title = fmt::format("{}", gui.info_message.title);
```
Comment (Zangetsu38) [inline vita3k/gui/src/gui.cpp:66]: For this, add const, and squash, no worry, no need keep my name, is your code 🫡

File: vita3k/gui/src/main_menubar.cpp (hunk 13 above)
Comment (Croden1999) [reply vita3k/gui/src/main_menubar.cpp:36]: > Yes, all time write check before menu 🫡, and set both is same line with &&

but actually we want create dir in app init, no need check exist

File: vita3k/gui/src/gui.cpp (hunk 14 above)
Comment (Zangetsu38) [inline vita3k/gui/src/gui.cpp:66]:
```suggestion
        const auto title = gui.info_message.title;
```
No need fmt format for this sorry 🤦‍♂️, and squash my commit, no need keep my name 👌

Comment (Croden1999) [reply vita3k/gui/src/gui.cpp:66]: i got it, sir

**Review by Macdu — APPROVED** [review]
